packages/generator/templates/app/app/auth/mutations/changePassword.ts

```diff
@@ -12,7 +12,11 @@ export default resolver.pipe(
     const user = await db.user.findFirst({ where: { id: ctx.session.userId as number } })
     if (!user) throw new NotFoundError()
 
-    await authenticateUser(user.email, currentPassword)
+   try {
+      await authenticateUser(user.email, currentPassword)
+    } catch (error: any) {
+      throw new Error("Invalid Password")
```

Comment:
Maybe let's throw `AuthenticationError` from `blitz`?
```
throw new AuthenticationError()
```

In a similar way as in `authenticateUser` function.

Reply:
The issue with the Authentication error is that actually logs the user out if they enter the wrong password within the change password form, and if they're accessing their dashboard then they should have already logged in.

Reply:
Okay I see. I'm not sure if we should return "invalid password" though. What if that error in `authenticateUser` didn't come from the password being invalid? E.g. if there was an error unrelated to the password or if email was incorrect or if there was an internal error (bug in our code, or something wrong with the database). Maybe we can have a new error instance (like AuthenticationError), e.g. InvalidPassword and throw it from `authenticateUser` explicitly?